t/cli/test_validate_config.sh

```diff
@@ -89,3 +89,60 @@ fi
 git checkout conf/config-default.yaml
 
 echo "passed: allow configuring node_listen as a number in the default config"
+
+# apisix test
+git checkout conf/config.yaml
+
+out=$(./bin/apisix test 2>&1 || true)
+if ! echo "$out" | grep "configuration test is successful"; then
+    echo "failed: configuration test should be successful"
+    exit 1
+fi
+
+echo "pass: apisix test"
+
+echo '
+nginx_config:
+    main_configuration_snippet: |
+        notexist on;
+' > conf/config.yaml
+
+out=$(./bin/apisix test 2>&1 || true)
+if ! echo "$out" | grep "configuration test failed"; then
+    echo "failed: should test failed when configuration invalid"
+    exit 1
+fi
+
+echo "passed: apisix test(failure scenario)"
+
+# apisix stop and restart
+git checkout conf/config.yaml
+
+./bin/apisix start
+sleep 1 # wait for apisix starts
+
+echo '
+nginx_config:
+    main_configuration_snippet: |
+        notexist on;
+' > conf/config.yaml
+
+out=$(./bin/apisix stop 2>&1 || true)
+if ! echo "$out" | grep "\[emerg\] unknown directive \"notexist\""; then
+    echo "failed: should stop failed when configuration invalid"
+    exit 1
+fi
+
+echo "passed: apisix stop"
+
+
+out=$(./bin/apisix stop 2>&1 || true)
+if ! (echo "$out" | grep "\[emerg\] unknown directive \"notexist\"") && ! (echo "$out" | grep "APISIX is running"); then
+    echo "failed: should restart failed when configuration invalid"
+    exit 1
+fi
+
+echo "passed: apisix restart"
+
+git checkout conf/config.yaml
```

Comment:
The configuration will be checked out after the test exits. We don't need to do it ourselves.